Gems/ROS2/Code/Source/Manipulation/JointsManipulationEditorComponent.cpp

```diff
@@ -0,0 +1,223 @@
+            {
+                AZ_Assert(false, "Joint names in hierarchy need to be unique (%s is not)!", jointName.c_str());
+                return;
+            }
+            AZ_Printf("JointsManipulationEditorComponent", "Adding joint info for hinge joint %s\n", jointName.c_str());
+            JointInfo jointInfo;
+            jointInfo.m_isArticulation = false;
+            jointInfo.m_axis = static_cast<PhysX::ArticulationJointAxis>(0);
+            jointInfo.m_entityComponentIdPair = idPair;
+            joints[jointName] = jointInfo;
+        }
+
+        bool TryGetFreeArticulationAxis(const AZ::EntityId& entityId, PhysX::ArticulationJointAxis& axis)
+        {
+            for (AZ::u8 i = 0; i <= static_cast<AZ::u8>(PhysX::ArticulationJointAxis::Z); i++)
+            {
+                PhysX::ArticulationJointMotionType type = PhysX::ArticulationJointMotionType::Locked;
+                axis = static_cast<PhysX::ArticulationJointAxis>(i);
+                // Use bus to prevent compilation error without PhysX Articulation support.
+                PhysX::ArticulationJointRequestBus::EventResult(type, entityId, &PhysX::ArticulationJointRequests::GetMotion, axis);
+                if (type != PhysX::ArticulationJointMotionType::Locked)
+                {
+                    return true;
+                }
+            }
+            return false;
+        }
+
+        void AddArticulationJointInfo(const AZ::EntityComponentIdPair& idPair, const AZStd::string& jointName, ManipulationJoints& joints)
+        {
+            PhysX::ArticulationJointAxis freeAxis;
+            bool hasFreeAxis = TryGetFreeArticulationAxis(idPair.GetEntityId(), freeAxis);
+            if (!hasFreeAxis)
+            { // Do not add a joint since it is a fixed one
+                AZ_Printf("JointsManipulationEditorComponent", "Articulation joint %s is fixed, skipping\n", jointName.c_str());
+                return;
+            }
+
+            if (joints.find(jointName) != joints.end())
+            {
+                AZ_Assert(false, "Joint names in hierarchy need to be unique (%s is not)!", jointName.c_str());
+                return;
+            }
+
+            AZ_Printf("JointsManipulationEditorComponent", "Adding joint info for articulation link %s\n", jointName.c_str());
+            JointInfo jointInfo;
+            jointInfo.m_isArticulation = true;
+            jointInfo.m_axis = freeAxis;
+            jointInfo.m_entityComponentIdPair = idPair;
+            joints[jointName] = jointInfo;
+        }
+
+        ManipulationJoints GetAllEntityHierarchyJoints(const AZ::EntityId& entityId)
+        { // Look for either Articulation Links or Hinge joints in entity hierarchy and collect them into a map.
+            // Determine kind of joints through presence of appropriate controller
+            bool supportsArticulation = false;
+            bool supportsClassicJoints = false;
+            JointsPositionControllerRequestBus::EventResult(
```

Comment:
How it can work ?
All handlers JointsPositionControllerRequestBus are GameComponent. Those are not activated in Editor mode.